ledger/src/shred/merkle.rs

```diff
@@ -951,18 +1053,44 @@ pub(super) fn make_shreds_from_data(
         .collect();
     // Generate coding shreds, populate merkle proof
     // for all shreds and attach signature.
-    let shreds: Result<Vec<_>, Error> = if shreds.len() <= 1 {
+    let shreds: Result<Vec<_>, Error> = if let Some(chained_merkle_root) = chained_merkle_root {
+        shreds
+            .into_iter()
+            .zip(next_code_index)
+            .scan(
+                chained_merkle_root,
+                |chained_merkle_root, (shreds, next_code_index)| {
+                    Some(
+                        make_erasure_batch(
+                            keypair,
+                            shreds,
+                            Some(*chained_merkle_root),
+                            next_code_index,
+                            is_last_in_slot,
+                            reed_solomon_cache,
+                        )
+                        .map(|(merkle_root, shreds)| {
+                            *chained_merkle_root = merkle_root;
+                            shreds
+                        }),
+                    )
+                },
+            )
+            .collect()
+    } else if shreds.len() <= 1 {
```

Comment:
is this case to avoid hitting the thread pool?

Reply:
`if shreds.len() <= 1` is to avoid threadpool when there is only one task, since in that case the threadpool will only add overhead.
(this is old code btw).